contracts/rewards/src/contract/query.rs

```diff
@@ -0,0 +1,240 @@
+use cosmwasm_std::{Storage, Uint64};
+use error_stack::Result;
+
+use crate::{
+    error::ContractError,
+    msg,
+    state::{self, Epoch, PoolId},
+};
+
+pub fn rewards_pool(
+    storage: &dyn Storage,
+    pool_id: PoolId,
+    block_height: u64,
+) -> Result<msg::RewardsPool, ContractError> {
+    let pool = state::load_rewards_pool(storage, pool_id.clone())?;
+    let current_params = state::load_params(storage);
+    let cur_epoch = Epoch::current(&current_params, block_height)?;
+
+    // the params could have been updated since the tally was created. Therefore we use the params from the
+    // active tally if it exists, otherwise we use the latest stored params.
+    let params = match state::load_epoch_tally(storage, pool_id.clone(), cur_epoch.epoch_num)? {
+        Some(epoch_tally) => epoch_tally.params,
+        None => current_params.params,
+    };
```

Comment:
Why do we want this? Don't we just want the latest params? I feel like a caller would prefer the latest info, instead of anything stale

Reply:
Actually, the caller probably most likely wants to know info for the current epoch, so I think what you have here is good.

Reply:
yes, the "current_params" might still be inactive if they didn't overwrite the epoch duration, so the way it's done here is correct